connector-agent-python/src/lib.rs

```diff
@@ -72,3 +72,40 @@ fn read_pg(py: Python, conn: &str, sqls: Vec<String>, schema: &str) -> PyResult<
         .collect();
     PyResult::Ok(ret.into_py_dict(py).to_object(py))
 }
+
+#[pyfunction]
+fn read_sql<'a>(
+    py: Python<'a>,
+    conn: &str,
+    query: &str,
+    col: &str,
+    min: i64,
+    max: i64,
+    num: i64,
+    return_type: &str,
+) -> PyResult<&'a PyAny> {
+    let mut queries: Vec<String> = vec![];
+    let partition_size = match (max - min + 1) % num == 0 {
+        true => (max - min + 1) / num,
+        false => (max - min + 1) / num + 1,
+    };
+
+    for i in 0..num {
+        let lower = min + i * partition_size;
+        let upper = min + (i + 1) * partition_size;
+        let partition_query = format!(
+            "{} where {} >= {} and {} < {}",
+            query, col, lower, col, upper
+        );
+        queries.push(partition_query);
+    }
+    let queries: Vec<_> = queries.iter().map(|s| s.as_str()).collect();
+    match return_type {
+        "pandas" => Ok(crate::pandas::write_pandas(py, conn, &queries, false)?),
+        "arrow" => todo!(),
```

Comment:
I think for this we can now throw a python exception indicating this not implemented.